sys/include/ztimer.h

```diff
@@ -488,11 +491,43 @@ void ztimer_update_head_offset(ztimer_clock_t *clock);
 void ztimer_init(void);
 
 /* default ztimer virtual devices */
+/**
+ * @brief   Base ztimer for the microsecond clock
+ */
+extern ztimer_clock_t *const ZTIMER_USEC_BASE;
+
 /**
  * @brief   Default ztimer microsecond clock
  */
 extern ztimer_clock_t *const ZTIMER_USEC;
 
+/**
+ * @def     ZTIMER_USEC_BASE_FREQ
+ *
+ * @brief   The frequency of the base ztimer for ZTIMER_USEC
+ */
+#ifdef XTIMER_HZ
```

Comment:
I see an issue here. I tried to keep all of the ztimer configuration in auto_init.c.
That means that the actual ztimer code compiles from the same headers. Including ztimer.h leads to the same on every platform.
Pulling the configuration here messes with that principle.

A solution would be to change the frequency defines to extern uint32 declarations, and define them in auto_init.c.

Reply:
I didn't use an `extern uint32_t` because then you can't do compile checks in other places, e.g. I can't know if I need to `shift` or scale down the timer at compile time, I would have to have a `ztimer_convert_shift_t ` and `ztimer_convert_frac_t` and at runtime check the speed to know which one to use.

Reply:
> I see an issue here. I tried to keep all of the ztimer configuration in auto_init.c.
> That means that the actual ztimer code compiles from the same headers. Including ztimer.h leads to the same on every platform.

Maybe we can have a configuration header? Similar to what we do for drivers with `driveXXX_params.h`?

Reply:
fine with me, as long as from ztimer itself, only auto_init.c includes it.

Reply:
`include/ztimer/config.h`?